fvm/src/call_manager/default.rs

```diff
@@ -349,7 +353,7 @@ where
                 store.data_mut().memory = memory;
 
                 // Lookup the invoke method.
-                let invoke: wasmtime::TypedFunc<(u32,), u32> = instance
+                let invoke: wasmtime::TypedFunc<(u32, u64, u32), u32> = instance
```

Comment:
Ouch, this is a massive breaking change for the pure sake of optimization. The reason why this is an _optimization_ is that this data can be otherwise obtained through a syscall.

If we are OK augmenting the invoke signature, the method number, caller, receiver, value received, etc. should totally take priority (ignoring momentarily that the method number is part of a _specific_ call convention, so it should technically not be an argument to invoke) over parameter metadata which may even be optional (such as if the method does not require params).

In a nutshell,. I really dislike this, but could stand behind putting these attributes in `InvocationContext`, fetcheable through a syscall that we know every single actor is going to make.

Reply:
Reverted.